On AVX2 cryptonite's own test suite crashes on unaligned access:
https://github.com/haskell-crypto/cryptonite/issues/347
--- a/cbits/decaf/include/constant_time.h
+++ b/cbits/decaf/include/constant_time.h
@@ -150,7 +150,7 @@ constant_time_lookup (
     for (j=0; j<n_table; j++, big_i-=big_one) {        
         big_register_t br_mask = br_is_zero(big_i);
         for (k=0; k<=elem_bytes-sizeof(big_register_t); k+=sizeof(big_register_t)) {
-            if (elem_bytes % sizeof(big_register_t)) {
+            if (1) {
                 /* unaligned */
                 ((unaligned_br_t *)(out+k))->unaligned
 			|= br_mask & ((const unaligned_br_t*)(&table[k+j*elem_bytes]))->unaligned;
